msg/Spawnable.msg

```diff
@@ -0,0 +1,5 @@
+# Robot or other object which can be spawned in simulation runtime.
+
+string uri                                # URI which will be accepted by SpawnEntity service.
+string description                        # Optional description for the user, e.g. "robot X with sensors A,B,C".
+SpawnableBounds spawn_bounds              # Optional spawn area bounds which fully encompass this object.
```

Comment:
In my experience this may require much more than just a single bounding box.

I would suggest starting off with:
* A list of discrete poses to sample from (you could use regular poses or `SpawnPose` above, e.g. "the robot could either be in the bedroom or kitchen centroid")
* A list of bounding boxes
* A list of polygons

And you could probably use standard messages for all of these, i.e., a `shape_msgs/SolidPrimitive` rather than a custom description of a box with lower and upper vertices.

Reply:
I am not sure if I understand your suggestion here. The spawn_bounds are used (optionally) to determine the space that the object would occupy, in a relative reference frame (to its top link), not where to spawn it (in the world).

I checked all the shape_msgs previously but none was fitting, including SolidPrimitive (e.g. asymmetrical box case is common / important as one other reviewer pointed out).

Reply:
> The spawn_bounds are used (optionally) to determine the space that the object would occupy, in a relative reference frame (to its top link), not where to spawn it (in the world)

Ah, I was understanding the latter. Never mind then!